app/controlplane/integrations/dependencytrack/cyclonedx/v1/integration.go

```diff
@@ -93,9 +114,65 @@ func (i *Integration) PreAttach(ctx context.Context, b *core.BundledConfig) (*co
 		return nil, fmt.Errorf("invalid attachment configuration: %w", err)
 	}
 
+	i.Logger.Infow("msg", "pre-attachment OK", "project", ar.GetConfig().GetProject())
+
 	return &core.PreAttachment{Configuration: ar.Config}, nil
 }
 
+// Send the SBOM to the configured Dependency Track instance
+func (i *DependencyTrack) Execute(ctx context.Context, opts *core.ExecuteReq) error {
```

Comment:
The new execution stage. Note how it already receives all the information it needs. No dependencies to credentials manager, downloader, nada.